src/Sintegra.php

```diff
@@ -51,38 +53,110 @@ public function get()
      */
     protected function totalize($sintegra)
     {
+        $numeroTotalRegistros = '0000000';
+        $numeroTotalRegistros = count(explode("\n", $sintegra));
+        $sintegraArray = explode("\n", $sintegra);
+        $numeroTotalRegistros += 1;
+        $numeroTotalRegistros = str_pad($numeroTotalRegistros, 8, "0", STR_PAD_LEFT);
+        $numeroTotalRegistros = str_pad($numeroTotalRegistros, 53, " ", STR_PAD_RIGHT);
+        $totalRegistros90 = '2';
+
+        $cnpj = substr($sintegraArray[0], 2, 14);
+        $cnpj = str_pad($cnpj, 14, " ", STR_PAD_RIGHT);
+        $ie = substr($sintegraArray[0], 16, 14);
+        $ie = str_pad($ie, 14, " ", STR_PAD_RIGHT);
+
         $tot = '';
         $keys = [];
-        $aefd = explode("\n", $sintegra);
-        foreach ($aefd as $element) {
-            $param = explode("|", $element);
-            if (!empty($param[1])) {
-                $key = $param[1];
+        $formatedSintegra = explode("\n", $sintegra);
+        foreach ($formatedSintegra as $element) {
+            $param = substr($element,0, 2);
+
+            if (!empty($param)) {
+                $key = $param;
                 if (!empty($keys[$key])) {
                     $keys[$key] += 1;
                 } else {
                     $keys[$key] = 1;
                 }
             }
         }
-        //Inicializa o bloco 9
-        $tot .= "|9001|0|\n";
-        $n = 0;
+
+        $totalRegistros10 = '00000000';
+        $totalRegistros11 = '00000000';
+        $totalRegistros50 = '00000000';
+        $totalRegistros51 = '00000000';
+        $totalRegistros53 = '00000000';
+        $totalRegistros54 = '00000000';
+        $totalRegistros60 = '00000000';
+        $totalRegistros61 = '00000000';
+        $totalRegistros70 = '00000000';
+        $totalRegistros71 = '00000000';
+        $totalRegistros74 = '00000000';
+        $totalRegistros75 = '00000000';
+        $totalRegistros85 = '00000000';
+        $totalRegistros88 = '00000000';
+        $totalRegistros86 = '00000000';
```

Comment:
Não precisa inicializar sendo que o str_pad já coloca 8 zeros que for varzio

Reply:
cara então como falei, se eu não deixar zerado, e não tiver o registro 50 por exemplo ele quebra já pelo fato de usar as posições, mesmo sem gerar estes registros é necessário ter isso.

Reply:
pq ele vai adicionar só se tiver a key se não vai sempre com = '00000000';